src/data_collection.py

```diff
@@ -153,34 +153,65 @@ def get_file_formats(files):
     return formats
 
 
-# NOTE: this function is temporary to test get_individual_metrics considering
-# there is not a main module that connects eerything yet
-def add_raw_data_to_json(path_to_repo, json_file_name):
-    """Check if raw data is in .json file and collects it if not."""
-    # get the data from .json file
-    current_data = json_handler.get_dict_from_json_file(json_file_name)
-    # check if data has not been collected and do so if it was not
-    if "RAW_DATA" not in current_data.keys():
-        print("Raw data has not been collected, collecting it now...")
-        # collects data from data_collection
-        raw_data = {"RAW_DATA": collect_commits_hash(path_to_repo)}
-        # Write raw data to .json file
+# This function simplifies gathering and writing raw data to json file
+# pylint: disable=C0330
+def colect_and_add_raw_data_to_json(
```

Comment:
would suggest collect_and_add be changed to "compile" or some other singular word.

Reply:
I wouldn't suggest using "compile" because it actually means something else than just "collect and add". It would be misleading if renamed to compile.

Reply:
I realize that the name is a bit long so I will try and come up with a better descriptive name